server/app/auth/CiviFormProfile.java

```diff
@@ -152,6 +151,8 @@ public CompletableFuture<Void> setAuthorityId(String authorityId) {
    * @return the future of the database operation
    */
   public CompletableFuture<Void> setEmailAddress(String emailAddress) {
+    profileData.setEmail(emailAddress);
```

Comment:
We have a number of cases where the below error ""Profile already contains an email address:" is thrown in SEA. What will happen to the profileData if an existing email address is added? Will the object be garbage collected?

Reply:
> What will happen to the profileData if an existing email address is added?

The profile's lifespan is the same as the session's, so if their session is destroyed the data stored in the profile will be gone with it.

Reply:
Line 161 raises ProfileMergeConflictException. Should profileData.setEmail() happen after it?

Reply:
I don't *think* that will change much since if there's an exception the whole session is probably broken, but there's no harming in setting it later on the idea we only update state anywhere after performing the check 👍🏻.